### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by anamariaUIC** [rapidsai/cudf#11981]
After executing:

`print(df.query('b == 3'))  `

`---------------------------------------------------------------------------
NvvmError                                 Traceback (most recent call last)
/tmp/ipykernel_237357/3669789841.py in <module>
----> 1 print(df.query('b == 3'))

/software/linux-el7-x86_64/apps/anaconda3/anaconda3/envs/rapids/lib/python3.7/site-packages/cudf/core/dataframe.py in query(self, expr, local_dict)
   3913             }
   3914             # Run query
-> 3915             boolmask = queryutils.query_execute(self, expr, callenv)
   3916             return self._apply_boolean_mask(boolmask)
   3917 

/software/linux-el7-x86_64/apps/anaconda3/anaconda3/envs/rapids/lib/python3.7/site-packages/cudf/utils/queryutils.py in query_execute(df, expr, callenv)
    228     # run kernel
    229     args = [out] + colarrays + envargs
--> 230     kernel.forall(nrows)(*args)
    231     out_mask = applyutils.make_aggregate_nullmask(df, columns=columns)
    232     return out.set_mask(out_mask).fillna(False)

/software/linux-el7-x86_64/compilers/python-3.7.3/lib/python3.7/site-packages/numba/cuda/compiler.py in __call__(self, *args)
    370             kernel = self.kernel
    371         else:
--> 372             kernel = self.kernel.specialize(*args)
    373         blockdim = self._compute_thread_per_block(kernel)
    374         griddim = (self.ntasks + blockdim - 1) // blockdim

/software/linux-el7-x86_64/compilers/python-3.7.3/lib/python3.7/site-packages/numba/cuda/compiler.py in specialize(self, *args)
    880         targetoptions['link'] = self.link
    881         specialization = Dispatcher(self.py_func, [types.void(*argtypes)],
--> 882                                     self._bind, targetoptions)
    883         self.specializations[cc, argtypes] = specialization
    884         return specialization

/software/linux-el7-x86_64/compilers/python-3.7.3/lib/python3.7/site-packages/numba/cuda/compiler.py in __init__(self, func, sigs, bind, targetoptions)
    806             if len(sigs) > 1:
    807                 raise TypeError("Only one signature supported at present")
--> 808             self.compile(sigs[0])
    809             self._can_compile = False
    810 

/software/linux-el7-x86_64/compilers/python-3.7.3/lib/python3.7/site-packages/numba/cuda/compiler.py in compile(self, sig)
    933             self.definitions[(cc, argtypes)] = kernel
    934             if self._bind:
--> 935                 kernel.bind()
    936             self.sigs.append(sig)
    937         return kernel

/software/linux-el7-x86_64/compilers/python-3.7.3/lib/python3.7/site-packages/numba/cuda/compiler.py in bind(self)
    574         Force binding to current CUDA context
    575         """
--> 576         self._func.get()
    577 
    578     @property

/software/linux-el7-x86_64/compilers/python-3.7.3/lib/python3.7/site-packages/numba/cuda/compiler.py in get(self)
    444         cufunc = self.cache.get(device.id)
    445         if cufunc is None:
--> 446             ptx = self.ptx.get()
    447 
    448             # Link

/software/linux-el7-x86_64/compilers/python-3.7.3/lib/python3.7/site-packages/numba/cuda/compiler.py in get(self)
    414             arch = nvvm.get_arch_option(*cc)
    415             ptx = nvvm.llvm_to_ptx(self.llvmir, arch=arch,
--> 416                                    **self._extra_options)
    417             self.cache[cc] = ptx
    418             if config.DUMP_ASSEMBLY:

/software/linux-el7-x86_64/compilers/python-3.7.3/lib/python3.7/site-packages/numba/cuda/cudadrv/nvvm.py in llvm_to_ptx(llvmir, **opts)
    523     cu.add_module(libdevice.get())
    524 
--> 525     ptx = cu.compile(**opts)
    526     # XXX remove debug_pubnames seems to be necessary sometimes
    527     return patch_ptx_debug_pubnames(ptx)

/software/linux-el7-x86_64/compilers/python-3.7.3/lib/python3.7/site-packages/numba/cuda/cudadrv/nvvm.py in compile(self, **options)
    230                                           for x in opts])
    231         err = self.driver.nvvmCompileProgram(self._handle, len(opts), c_opts)
--> 232         self._try_error(err, 'Failed to compile\n')
    233 
    234         # get result

/software/linux-el7-x86_64/compilers/python-3.7.3/lib/python3.7/site-packages/numba/cuda/cudadrv/nvvm.py in _try_error(self, err, msg)
    248 
    249     def _try_error(self, err, msg):
--> 250         self.driver.check_error(err, "%s\n%s" % (msg, self.get_log()))
    251 
    252     def get_log(self):

/software/linux-el7-x86_64/compilers/python-3.7.3/lib/python3.7/site-packages/numba/cuda/cudadrv/nvvm.py in check_error(self, error, msg, exit)
    138                 sys.exit(1)
    139             else:
--> 140                 raise exc
    141 
    142 

NvvmError: Failed to compile

<unnamed> (63, 19): parse expected comma after load's type
NVVM_ERROR_COMPILATION`

cuDF Version: 21.12.02

**Describe the bug**

**Steps/Code to reproduce bug**
Follow this guide http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

**Expected behavior**

**Environment overview (please complete the following information)**
 - Environment location: [Bare-metal, Docker, Cloud(specify cloud provider)]
 - Method of cuDF install: [conda, Docker, or from source]
   - If method of install is [Docker], provide `docker pull` & `docker run` commands used

**Environment details**
Please run and paste the output of the `cudf/print_env.sh` script here, to gather any other relevant environment details

**Additional context**

**Comment by anamariaUIC** [comment]
Additionally this doesn't work anymore either:

`def add_ten(num):
    return num + 10

print(s.applymap(add_ten))`

NvvmError: Failed to compile

<unnamed> (20, 18): parse expected comma after load's type
NVVM_ERROR_COMPILATION

What is new syntax for this?

**Comment by vyasr** [comment]
Hi @anamariaUIC it looks like you pasted your traceback above the issue form but didn't fill out the fields themselves. Could you go back through and add the information requested in the form ("Describe the bug", "Steps to Reproduce", "Environment Overview", etc)? That will help us to diagnose the problem.

You say "this doesn't work anymore", did it used to work? If so, what changed? Did you update cudf, or numba, or something else?

**Comment by anamariaUIC** [comment]
this did work in April 2022. I didn't update any libraries.

Steps to reproduce:

`df = cudf.DataFrame({'a': list(range(20)),
'b': list(reversed(range(20))),
'c': list(range(20))})`

`print(df.query('b == 3')) `

Please let me know if you need more info.

**Comment by anamariaUIC** [comment]
I am also seeing error when I run this cell:

`df = df.apply_rows(haversine_distance_kernel,
                   incols=['lat1', 'lon1', 'lat2', 'lon2'],
                   outcols=dict(out=np.float64),
                   kwargs=dict())`

From [this notebook](https://github.com/Quansight/scipy-2019-rapids-tutorial/blob/master/cudf/Intro%20to%20cuDF%20UDFs.ipynb)

error is:

`NvvmError: Failed to compile

<unnamed> (128, 19): parse expected comma after load's type
NVVM_ERROR_COMPILATION`

**Comment by beckernick** [comment]
The cuDF 21.12 packages haven't changed since patch release 21.12.02 [10 months and 8 days ago](https://anaconda.org/rapidsai/cudf/files). This release pins `numba >=0.54`, so it's conceptually possible that there is an unexpected interaction (but this feels less likely).

Has your system perhaps changed between April and now? Could you clone this repository and run the `print_env.sh` script provided to gather relevant environment details, as suggested in the issue prompt?

**Comment by anamariaUIC** [comment]
yes I will do that. In the meantime can you please confirm that you're able to run [this notebook](https://github.com/Quansight/scipy-2019-rapids-tutorial/blob/master/cudf/Intro%20to%20cuDF%20UDFs.ipynb) without errors and if yes which version of libraries you had?
